C++/Toolbox/Sink/ThreadSafeProxySink.h

```diff
@@ -0,0 +1,69 @@
+#ifndef _TOOLBOX_SINK_THREAD_SAFE_PROXY_SINK_H_
+#define _TOOLBOX_SINK_THREAD_SAFE_PROXY_SINK_H_
+
+#include <Toolbox/Sink/BasicSink.h>
+#include <Toolbox/Sink/ThreadSafeSink.h>
+
+#include <memory>
+#include <type_traits>
+
+namespace Toolbox::Sink
+{
+
+template <typename SinkType,
+          typename OutputType = typename SinkType::value_type,
+          typename = typename std::enable_if<std::is_base_of<BasicSink<OutputType>, SinkType>::value>::type>
+class ThreadSafeProxySink : public ThreadSafeSink<OutputType>
+{
+public:
+ThreadSafeProxySink (SinkType sink)
+: _sink (std::move (sink))
+{
+
+}
+
+ThreadSafeProxySink (ThreadSafeProxySink && sink)
+: _sink (std::move (sink._sink))
+{
+
+}
+
+ThreadSafeProxySink & operator= (ThreadSafeProxySink && sink)
```

Comment:
Regarding both the copy constructor and copy assignment operator: I'm not sure of the best practices here, but does it seem more natural to you @DannyPeck to leverage the `default` and `delete` keywords here? You are effectively converting your copy constructor into a move constructor and your copy assignment operator into a move assignment operator. Why not simply `delete` the copy constructor/assignment and either define or `default` the move constructor/assignment?

Reply:
That is actually the Move Constructor and Move Assignment Operator.  And interestingly, if I switch to using the default key word for those methods it tries to use the copy constructor and it doesn't work.  Also the copy constructors are explicitly deleted in the BasicSink base class so they are implicitly deleted in all subclasses.  So in conclusion, that is exactly what I did.  Is my signature for the move constructor wrong?

Reply:
Yeah now that I look at this...I think I was just really tired. You're totally correct. This is the move constructor/assignment operator. Your syntax is correct. My bad 😴

Reply:
You're good!  I still think it's weird that I had to define it and that default didn't work...